src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,48 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<K, V> {
+    private K first;
+    private V second;
+
+    private Pair(K first, V second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <K, V> Pair<K, V> of(K first, V second) {
+        return new Pair(first, second);
+    }
+
+    public K getFirst() {
+        return this.first;
+    }
+
+    public V getSecond() {
+        return this.second;
```

Comment:
do we need to use `this`?

Reply:
It was in task:

Implement the getFirst(), getSecond(), equals(), and hashCode() methods, as well as the static factory method of(). The constructor must be private.